syntax/std_seq.go

```diff
@@ -64,5 +65,108 @@ func stdSeq() rel.Attr {
+			switch args[1].(type) {
+			case rel.String:
+				return rel.NewBool(strings.HasPrefix(mustAsString(args[1]), mustAsString(args[0])))
+			case rel.Array:
+				return arrayPrefix(args[1].(rel.Array), args[0])
+			case rel.Bytes:
+				return rel.NewBool(strings.HasPrefix(args[1].String(), args[0].String()))
+			}
+
+			return rel.NewBool(false)
+		}),
+		createNestedFuncAttr("has_suffix", 2, func(args ...rel.Value) rel.Value {
+			switch args[1].(type) {
+			case rel.String:
+				return rel.NewBool(strings.HasSuffix(mustAsString(args[1]), mustAsString(args[0])))
+			case rel.Array:
+				return arraySuffix(args[1].(rel.Array), args[0])
+			case rel.Bytes:
+				return rel.NewBool(strings.HasSuffix(args[1].String(), args[0].String()))
+			}
+
+			return rel.NewBool(false)
+		}),
+		createNestedFuncAttr("sub", 3, func(args ...rel.Value) rel.Value {
+			switch args[2].(type) {
+			case rel.String:
+				return rel.NewString(
+					[]rune(
+						strings.ReplaceAll(
+							mustAsString(args[2]),
+							mustAsString(args[0]),
+							mustAsString(args[1]),
+						),
+					),
+				)
+			case rel.Array:
+				return arraySub(args[2].(rel.Array), args[0], args[1])
+			case rel.Bytes:
+				return rel.NewBytes([]byte(strings.ReplaceAll(args[2].String(), args[0].String(), args[1].String())))
+			}
+
+			panic(fmt.Errorf("expected subject sequence types are %s, %s and %s, but the actual type is %s",
+				reflect.TypeOf(rel.String{}), reflect.TypeOf(rel.Array{}), reflect.TypeOf(rel.Bytes{}),
+				reflect.TypeOf(args[2])))
+		}),
+		createNestedFuncAttr("split", 2, func(args ...rel.Value) rel.Value {
+			switch args[1].(type) {
+			case rel.String:
+				splitted := strings.Split(mustAsString(args[1]), mustAsString(args[0]))
+				vals := make([]rel.Value, 0, len(splitted))
+				for _, s := range splitted {
+					vals = append(vals, rel.NewString([]rune(s)))
+				}
+				return rel.NewArray(vals...)
+			case rel.Array:
+				return arraySplit(args[1].(rel.Array), args[0])
+			case rel.Bytes:
+				return bytesSplit(args[1].(rel.Bytes), args[0])
+			}
+
+			panic(fmt.Errorf("expected subject sequence types are %s, %s and %s, but the actual type is %s",
+				reflect.TypeOf(rel.String{}), reflect.TypeOf(rel.Array{}), reflect.TypeOf(rel.Bytes{}),
+				reflect.TypeOf(args[2])))
+		}),
+		createNestedFuncAttr("join", 2, func(args ...rel.Value) rel.Value {
+			switch a1 := args[1].(type) {
+			case rel.Array:
+				switch a1.Values()[0].(type) {
+				case rel.String:
+					return strJoin(args...)
+				case rel.Value:
+					return arrayJoin(a1, args[0])
+				}
+			case rel.Bytes:
+				return bytesJoin(args[1].(rel.Bytes), args[0].(rel.Bytes))
+			}
+
+			panic(fmt.Errorf("expected subject sequence types are %s and %s, but the actual type is %s",
+				reflect.TypeOf(rel.Array{}), reflect.TypeOf(rel.Bytes{}),
+				reflect.TypeOf(args[2])))
+		}),
 	)
 }
+
+func strJoin(args ...rel.Value) rel.Value {
+	strs := args[1].(rel.Set)
```

Comment:
this should be casted to `rel.Array`

Reply:
It is a set in case `//seq.join(",",[])`.